Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm

```diff
@@ -1201,6 +1201,10 @@ - (void)_holdWindowResizeSnapshotWithReason:(NSString *)reason
     if (!self.window || ![self.window respondsToSelector:@selector(_holdResizeSnapshotWithReason:)])
         return;
 
+    // Only create the handler if entering or exiting window full screen, and not element full screen.
+    if ([self fullscreenState] != WKFullscreenStateNotInFullscreen)
+        return;
```

Comment:
I have a slight preference for putting this condition in `_web_windowWillEnterFullScreen` and `_web_windowWillExitFullScreen`, since `_holdWindowResizeSnapshotWithReason` doesn't necessarily know or need to care about fullscreen itself.

Might be good to get a second opinion on that.